### ISSUE TITLE
Network is unable to converge with C-triplet loss

### ISSUE THREAD

**Issue by inlmouse** [happynear/NormFace#17]
My network works well with C-contrastive loss. However, when I switch to C-triplet loss without any network structure modification, it cannot converge. The loss will keep around 1.0. Some attempts such as changed the parameter *margin* and *negative_weight* have been made. But you know, still doesn't work...

Is there any trick or something that you did, which is able to make the net converge with C-triplet loss?

**Comment by happynear** [comment]
I train the C-triplet loss using these parameters: https://github.com/happynear/NormFace/blob/master/prototxt/normalized_Euclidean_triplet.prototxt .
 You may have a try.

**Comment by inlmouse** [comment]
Unfortunately, I use your default parameters at the beginning. It cannot converge if I train the net from random initiation or softmax pre-trained model. I even tried fine-turn the net from a converged model which used C-constrastive loss. It has a good start: the loss can be less than 0.25. But with the training progressed, the loss get more and more higher and the accuracy goes 0 within 10000 iteration. So, any idea?

**Comment by happynear** [comment]
Nothing special is needed. The only thing left is the learning rate. When fine-tuning all my proposed losses, the learning rate should be small, say 0.001. But since you have successfully trained C-contrastive loss, I think you already followed this setting. 

Well, I don't know the reason now. I also came across difficulty when training C-triplet loss. Then I added the hard mining parameter and it could converge. Anyway, the key factor of these loss functions is the normalization. All loss functions can promote the performance after using the normalization operation. I think you don't need to pay much attention to any specific loss functions.
